code/game/gamemodes/vampire/vampire_powers.dm

```diff
@@ -546,6 +546,21 @@
 			heal_organ_damage(50, 50)
 			blood_used += 12
 
+		for (var/A in src.organs)
+			var/obj/item/organ/external/E = A
+			for (var/X in E.wounds)
+				var/datum/wound/W = X
+				if (W && W.internal)
+					E.wounds -= W
+					blood_used += 12
+					break
```

Comment:
This will potentially fix all wounds, if no organs are actually set to broken.
```dm
		for (var/A in organs)
			var/healed = FALSE
			var/obj/item/organ/external/E = A
			for (var/X in E.wounds)
				var/datum/wound/W = X
				if (W && W.internal)
					E.wounds -=W
					blood_used += 12
					healed = TRUE
					break

			if (E.status & ORGAN_BROKEN)
				E.status &= ~ORGAN_BROKEN
				E.stage = 0
				blood_used += 12
				healed = TRUE

			if (healed)
				break
```

The healing loop is called often enough for one severe-heal per cycle to be sufficient.